### PR TITLE
Added stamina and sprint. Currently, does not work.

### PR DESCRIPTION
**Pull request by Fedic**
Need help from you guys. StartSprinting()/StopSprinting() do not get called so there is something wrong with the bindings maybe?
image
image

### CODE REVIEW

File: Source/Redemption/Private/RedemptionPlayer.cpp [hunk 1]
```diff
@@ -173,3 +221,25 @@ void ARedemptionPlayer::EndCrouch()
...
+		bIsSprinting = true;
+		GetCharacterMovement()->MaxWalkSpeed = MaxSprintSpeed;
+	}
+}
+
+void ARedemptionPlayer::StopSprinting()
+{
+	UE_LOG(LogTemp, Warning, TEXT("StopSprinting called"));
+	bIsSprinting = false;
+	GetCharacterMovement()->MaxWalkSpeed = 500.f;
```
Comment (mac2648) [inline Source/Redemption/Private/RedemptionPlayer.cpp:239]: I think the player moves slower than this normally

File: Source/Redemption/Private/RedemptionPlayer.cpp [hunk 2]
```diff
@@ -32,6 +40,11 @@ ARedemptionPlayer::ARedemptionPlayer()
 		{
 			WidgetComponent->SetWidgetClass((WidgetClass.Class));
 		}
+		static ConstructorHelpers::FObjectFinder<UInputAction> SprintActionFinder(TEXT("/Game/InputActions/IA_Sprint"));
+		if (SprintActionFinder.Succeeded())
+		{
+			SprintAction = SprintActionFinder.Object;
+		}
```
Comment (mac2648) [inline Source/Redemption/Private/RedemptionPlayer.cpp:47]: I think the problem is here, are you sure this path will get you the IA_Sprint?

Comment (Fedic) [reply Source/Redemption/Private/RedemptionPlayer.cpp:47]: I fixed that and changed it to "/Game/Player/Input/IA_Sprint" but still nothing is getting called.

Comment (mac2648) [reply Source/Redemption/Private/RedemptionPlayer.cpp:47]: I will download it on my end and see if I can figure it out

Comment (mac2648) [reply Source/Redemption/Private/RedemptionPlayer.cpp:47]: on the debug I saw it was null, if you check the path you will see that it is "/Game/Player/Input/Actions/IA_Sprint"

File: Source/Redemption/Private/RedemptionPlayer.cpp [hunk 3]
```diff
@@ -21,6 +21,14 @@ ARedemptionPlayer::ARedemptionPlayer()
 {
 	HealthComp = CreateDefaultSubobject<UHealthComponent>(TEXT("Health Comp"));
 	PrimaryActorTick.bCanEverTick = true;
+	// Stamina settings
+	MaxStamina = 100.0f;
+	Stamina = MaxStamina;
+	MaxSprintSpeed = 800.0f; // How fast you run
```
Comment (mac2648) [inline Source/Redemption/Private/RedemptionPlayer.cpp:27]: I think 800 is to fast for your game no?

File: Source/Redemption/Private/RedemptionPlayer.cpp [hunk 4]
```diff
@@ -173,3 +222,25 @@ void ARedemptionPlayer::EndCrouch()
...
+		bIsSprinting = true;
+		GetCharacterMovement()->MaxWalkSpeed = MaxSprintSpeed;
+	}
+}
+
+void ARedemptionPlayer::StopSprinting()
+{
+	UE_LOG(LogTemp, Warning, TEXT("StopSprinting called"));
+	bIsSprinting = false;
+	GetCharacterMovement()->MaxWalkSpeed = 500.f;
```
Comment (mac2648) [inline Source/Redemption/Private/RedemptionPlayer.cpp:240]: Screenshot 2024-03-02 153821
this is the player normal speed

**Review by mac2648 — CHANGES_REQUESTED** [review]

**Review by mac2648 — APPROVED** [review]

**Review by thakkarnandish — APPROVED** [review]

**Pull request merged**